fastly/block_fastly_service_v1_logging_sftp.go

```diff
@@ -0,0 +1,289 @@
+				Description: "The unique name of the SFTP logging endpoint",
+			},
+
+			"address": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The SFTP address to stream logs to.",
+			},
+
+			"user": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The username for the server",
+			},
+
+			"path": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The path to upload log files to. If the path ends in / then it is treated as a directory.",
+			},
+
+			"ssh_known_hosts": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "A list of host keys for all hosts we can connect to over SFTP.",
+			},
+
+			// Optional fields
+			"port": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     22,
+				Description: "The port the SFTP service listens on. (Default: 22)",
+			},
+
+			"password": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The password for the server. If both password and secret_key are passed, secret_key will be preferred.",
+				Sensitive:   true,
+			},
+
+			"secret_key": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The SSH private key for the server. If both password and secret_key are passed, secret_key will be preferred.",
+				Sensitive:   true,
+			},
+
+			"public_key": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "A PGP public key that Fastly will use to encrypt your log files before writing them to disk.",
+			},
+
+			"period": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     3600,
+				Description: "How frequently log files are finalized so they can be available for reading (in seconds, default 3600).",
+			},
+
+			"gzip_level": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     0,
+				Description: "What level of GZIP encoding to have when dumping logs (default 0, no compression).",
+			},
+
+			"timestamp_format": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "%Y-%m-%dT%H:%M:%S.000",
+				Description: "The strftime specified timestamp formatting (default `%Y-%m-%dT%H:%M:%S.000`).",
+			},
+
+			"format": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Apache-style string or VCL variables to use for log formatting.",
+			},
+
+			"format_version": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      2,
+				Description:  "The version of the custom logging format used for the configured endpoint. Can be either 1 or 2. (default: 2)",
+				ValidateFunc: validateLoggingFormatVersion(),
+			},
+
+			"placement": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Where in the generated VCL the logging call should be placed",
+				ValidateFunc: validateLoggingPlacement(),
+			},
+
```

Comment:
@mccurdyc TODO: update to use new go-fastly version added in (https://github.com/terraform-providers/terraform-provider-fastly/pull/248) and add MessageType field.

Reply:
done